packages/Webkul/CartRule/src/Http/Controllers/CartRuleController.php

```diff
@@ -169,7 +169,7 @@ public function update(Request $request, $id)
             'customer_groups'     => 'required|array|min:1',
             'coupon_type'         => 'required',
             'use_auto_generation' => 'required_if:coupon_type,==,1',
-            'coupon_code'         => 'required_if:use_auto_generation,==,0',
+            'coupon_code'         => 'required_if:use_auto_generation,==,0|unique:cart_rule_coupons,code',
```

Comment:
Will not work for the update. Change this to 'required_if:use_auto_generation,==,0|unique:cart_rule_coupons,code' . $id

Reply:
ID added.